### PR TITLE
KEP-4601: alpha docs

### PR DESCRIPTION
**Pull request by liggitt**
Alpha docs for https://github.com/kubernetes/enhancements/issues/4601

### CODE REVIEW

**Comment by MaryamTavakkoli** [comment]
Hello  @liggitt👋  Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before **Tuesday July 16th 2024 18:00 PST.** Thank you!

**Comment by MaryamTavakkoli** [comment]
Hello @liggitt👋 Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday July 16th 2024 18:00 PST. Thank you!

**Comment by Princesso** [comment]
Hi @liggitt , a gentle reminder that tomorrow is the deadline for having your Docs PR ready for review. Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before tomorrow, Tuesday, July 16th, 2024 18:00 PST.

**Comment by liggitt** [comment]
Content is updated for review

/assign @deads2k @enj

File: content/en/docs/reference/access-authn-authz/node.md [hunk 1]
```diff
@@ -27,6 +27,11 @@ Read operations:
 * secrets, configmaps, persistent volume claims and persistent volumes related
   to pods bound to the kubelet's node
 
+{{< feature-state state="alpha" for_k8s_version="v1.31" feature_gate_name="AuthorizeNodeWithSelectors" >}}
```
Comment (tengqm) [inline content/en/docs/reference/access-authn-authz/node.md:30]:
```suggestion
{{< feature-state feature_gate_name="AuthorizeNodeWithSelectors" >}}
```

File: content/en/docs/reference/access-authn-authz/webhook.md [hunk 2]
```diff
@@ -164,13 +164,50 @@ Access to non-resource paths are sent as:
 }
 ```
 
+{{< feature-state state="alpha" for_k8s_version="v1.31" feature_gate_name="AuthorizeWithSelectors" >}}
```
Comment (tengqm) [inline content/en/docs/reference/access-authn-authz/webhook.md:167]:
```suggestion
{{< feature-state feature_gate_name="AuthorizeWithSelectors" >}}
```

File: content/en/docs/reference/using-api/cel.md [hunk 3]
```diff
@@ -200,7 +200,18 @@ To perform an authorization check for a service account:
 | `authorizer.serviceAccount('default', 'myserviceaccount').resource('deployments').check('delete').allowed()` | Checks if the service account is authorized to delete deployments. |
 {{< /table >}}
 
+{{< feature-state state="alpha" for_k8s_version="v1.31" >}}
```
Comment (tengqm) [inline content/en/docs/reference/using-api/cel.md:203]:
```suggestion
{{< feature-state feature_gate_name="AuthorizeWithSelectors"  >}}
```

Comment (liggitt) [reply content/en/docs/reference/using-api/cel.md:203]: does that pull in the version info?

Comment (tengqm) [reply content/en/docs/reference/using-api/cel.md:203]: Yes, it does.

**Comment by MaryamTavakkoli** [comment]
Hello @liggitt👋! I'm reaching out from the Docs team. Just checking in as we approach Docs Freeze on Tuesday July 30th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as lgtm and approve labels applied, without any unaddressed comments or concerns from SIG Docs. The status of this enhancement is marked as at risk for docs freeze. Thank you!

File: content/en/docs/reference/access-authn-authz/node.md (hunk 1 above)
Comment (tengqm) [reply content/en/docs/reference/access-authn-authz/node.md:30]: you don't need to add 'state' or 'for_k8s_version' for this shortcode.

File: content/en/docs/reference/access-authn-authz/node.md [hunk 4]
```diff
@@ -27,6 +27,11 @@ Read operations:
 * secrets, configmaps, persistent volume claims and persistent volumes related
   to pods bound to the kubelet's node
 
+{{< feature-state feature_gate_name="AuthorizeNodeWithSelectors" >}}
+
+When the `AuthorizeNodeWithSelectors` feature is enabled, kubelets are only allowed to
```
Comment (deads2k) [inline content/en/docs/reference/access-authn-authz/node.md:32]: is it just this gate or both gates required?

**Comment by deads2k** [comment]
one question, lgtm otherwise.

File: content/en/docs/reference/access-authn-authz/node.md (hunk 4 above)
Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/node.md:32]: startup validation won't _let_ you enable AuthorizeNodeWithSelectors without also enabling AuthorizeWithSelectors, tried to keep this doc focused on the specific gate enabling this capability

**Comment by sftim** [comment]
If a review is needed @MaryamTavakkoli, the folks to ask are @kubernetes/sig-docs-en-reviews

File: content/en/docs/reference/access-authn-authz/node.md (hunk 4 above)
Comment (ritazh) [reply content/en/docs/reference/access-authn-authz/node.md:32]: +1 for adding a short note about enabling the other feature gate is needed to save users time to figure out why the feature isn't working

File: content/en/docs/reference/access-authn-authz/webhook.md [hunk 5]
```diff
@@ -164,13 +164,50 @@ Access to non-resource paths are sent as:
 }
 ```
 
+{{< feature-state feature_gate_name="AuthorizeWithSelectors" >}}
+
+With the `AuthorizeWithSelectors` feature enabled, field and label selectors in the request
+are passed to the authorization webhook. The webhook can make authorization decisions
+informed by the scoped field and label selectors, if it wishes.
```
Comment (enj) [inline content/en/docs/reference/access-authn-authz/webhook.md:171]: IMO this should more strongly encourage webhook authors to read the specifc SAR docs, i.e. CVE-2022-2880.

File: content/en/docs/reference/command-line-tools-reference/feature-gates/authorize-with-selectors.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+stages:
+  - stage: alpha
+    defaultValue: false
+    fromVersion: "1.31"
+---
+Allows authorization to use field and label selectors.
+Enables `fieldSelector` and `labelSelector` fields in the [SubjectAccessReview API](/docs/reference/kubernetes-api/authorization-resources/subject-access-review-v1/),
+passes field and label selector information to [authorization webhooks](/docs/reference/access-authn-authz/webhook/),
+enables `fieldSelector` and `labelSelector` functions in the [authorizer CEL library](https://pkg.go.dev/k8s.io/apiserver/pkg/cel/library#AuthzSelectors),
+and enables checking `fieldSelector` and `labelSelector` fields in [authorization webhook `matchConditions`](/docs/reference/access-authn-authz/authorization/#using-configuration-file-for-authorization).
```
Comment (enj) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/authorize-with-selectors.md:17]: Do we have an example with CEL+selector in `matchConditions`?

Comment (liggitt) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/authorize-with-selectors.md:17]: no example directly in the matchConditions, but it references the subjectaccessreview API doc which would now include these fields

File: content/en/docs/reference/access-authn-authz/webhook.md (hunk 5 above)
Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/webhook.md:171]: done

File: content/en/docs/reference/access-authn-authz/node.md (hunk 4 above)
Comment (liggitt) [reply content/en/docs/reference/access-authn-authz/node.md]: done

**Pull request merged**